From 47c6706338cedbba7a5fd3f6d25ec407d161db7c Mon Sep 17 00:00:00 2001
From: Ian Lin <ian.lin-ee@infineon.com>
Date: Thu, 6 Jan 2022 21:15:23 -0600
Subject: [PATCH 112/179] brcmfmac: skip 6G oob scan report

WAR to skip 6G oob scan report before host could support 6E.

---
 .../net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c   | 8 ++++++++
 .../net/wireless/broadcom/brcm80211/include/brcmu_d11.h   | 2 +-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
index 8ef6032b37c4..9bc23fbc3c92 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
@@ -3670,6 +3670,14 @@ brcmf_cfg80211_escan_handler(struct brcmf_if *ifp,
 		}
 		bss_info_le = &escan_result_le->bss_info_le;
 
+		/* WAR to skip 6G scan results before 6E support is complete in host */
+		if ((bss_info_le->chanspec & BRCMU_CHSPEC_D11AC_BND_MASK) ==
+			BRCMU_CHSPEC_D11AC_BND_6G) {
+			bphy_err(drvr, "Currently skip 6G scan, chanspec 0x%04x\n",
+				 bss_info_le->chanspec);
+			goto exit;
+		}
+
 		if (brcmf_p2p_scan_finding_common_channel(cfg, bss_info_le))
 			goto exit;
 
diff --git a/drivers/net/wireless/broadcom/brcm80211/include/brcmu_d11.h b/drivers/net/wireless/broadcom/brcm80211/include/brcmu_d11.h
index f6344023855c..9a54a046a795 100644
--- a/drivers/net/wireless/broadcom/brcm80211/include/brcmu_d11.h
+++ b/drivers/net/wireless/broadcom/brcm80211/include/brcmu_d11.h
@@ -82,7 +82,7 @@
 #define BRCMU_CHSPEC_D11AC_BND_SHIFT	14
 #define  BRCMU_CHSPEC_D11AC_BND_2G	0x0000
 #define  BRCMU_CHSPEC_D11AC_BND_3G	0x4000
-#define  BRCMU_CHSPEC_D11AC_BND_4G	0x8000
+#define  BRCMU_CHSPEC_D11AC_BND_6G	0x8000
 #define  BRCMU_CHSPEC_D11AC_BND_5G	0xc000
 
 #define BRCMU_CHAN_BAND_2G		0
-- 
2.17.1

